x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx

```diff
@@ -36,108 +33,28 @@ function EmbeddedMapComponent({
+  const [layerList, setLayerList] = useState<LayerDescriptor[]>([]);
 
-  const embeddableRoot: React.RefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
-
-  const {
-    embeddable: embeddablePlugin,
-    maps,
-    notifications,
-  } = useKibana<ApmPluginStartDeps>().services;
-
-  useEffect(() => {
-    async function setupEmbeddable() {
-      const factory = embeddablePlugin?.getEmbeddableFactory<
-        MapEmbeddableInput,
-        MapEmbeddableOutput,
-        MapEmbeddable
-      >(MAP_SAVED_OBJECT_TYPE);
-
-      if (!factory) {
-        setError(true);
-        notifications?.toasts.addDanger({
-          title: i18n.translate('xpack.apm.serviceOverview.embeddedMap.error.toastTitle', {
-            defaultMessage: 'An error occurred when adding map embeddable',
-          }),
-          text: i18n.translate('xpack.apm.serviceOverview.embeddedMap.error.toastDescription', {
-            defaultMessage: `Embeddable factory with id "{embeddableFactoryId}" was not found.`,
-            values: {
-              embeddableFactoryId: MAP_SAVED_OBJECT_TYPE,
-            },
-          }),
-        });
-        return;
-      }
-
-      const input: MapEmbeddableInput = {
-        attributes: { title: '' },
-        id: uuidv4(),
-        title: i18n.translate('xpack.apm.serviceOverview.embeddedMap.input.title', {
-          defaultMessage: 'Latency by country',
-        }),
-        filters,
-        viewMode: ViewMode.VIEW,
-        mapCenter: { lat: 20.43425, lon: 0, zoom: 1.25 },
-        isLayerTOCOpen: false,
-        query: {
-          query: kuery,
-          language: 'kuery',
-        },
-        timeRange: {
-          from: start,
-          to: end,
-        },
-        hideFilterActions: true,
-      };
-
-      const embeddableObject = await factory.create(input);
-
-      setEmbeddable(embeddableObject);
-    }
-
-    setupEmbeddable();
-    // Set up exactly once after the component mounts
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  // We can only render after embeddable has already initialized
-  useEffect(() => {
-    if (embeddableRoot.current && embeddable) {
-      embeddable.render(embeddableRoot.current);
-    }
-  }, [embeddable, embeddableRoot]);
+  const { maps } = useKibana<ApmPluginStartDeps>().services;
 
   useEffect(() => {
-    const setLayerList = async () => {
-      if (embeddable && !isErrorEmbeddable(embeddable) && dataView?.id) {
-        const layerList = await getLayerList({
-          selectedMap,
-          maps,
-          dataViewId: dataView.id,
-        });
-        await Promise.all([embeddable.setLayerList(layerList), embeddable.reload()]);
+    const updateLayers = async () => {
+      if (dataView?.id) {
+        try {
+          const layers = await getLayerList({
+            selectedMap,
+            maps,
+            dataViewId: dataView.id,
+          });
+          setLayerList(layers);
+        } catch (e) {
+          setError(true);
+        }
```

Comment:
nit, async task is not canceled and could return incorrect results if function is results return out of order. How about returning a callback, like the example below, that cancels the results from the previous call.

```
let ignore = false;

const results = await asyncStuff();
if (!ignore) {
  setLayerList(results);
}
return ()=> {
    ignore = true;
}
```

Reply:
I was able to remove this `useEffect` in https://github.com/elastic/kibana/pull/182753/commits/c2cf3a874f62f842fb1207ca32f0434d8b629ae4 since there is no need to asynchronously load the basemap descriptor because the `MapComponent` already provides it.